internal/services/mssql/mssql_database_resource.go

```diff
@@ -488,133 +486,170 @@ func resourceMsSqlDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) erro
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.DatabaseID(d.Id())
+	id, err := commonids.ParseDatabaseID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	serverId := parse.NewServerID(id.SubscriptionId, id.ResourceGroup, id.ServerName)
+	serverId := commonids.NewSqlServerID(id.SubscriptionId, id.ResourceGroupName, id.ServerName)
+	d.Set("server_id", serverId.ID())
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
+	resp, err := client.Get(ctx, pointer.From(id), databases.DefaultGetOperationOptions())
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			d.SetId("")
 			return nil
 		}
 		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
 
-	d.Set("name", resp.Name)
-	d.Set("server_id", serverId.ID())
-
+	geoBackupPolicy := true
 	skuName := ""
 	ledgerEnabled := false
-	if props := resp.DatabaseProperties; props != nil {
-		d.Set("auto_pause_delay_in_minutes", props.AutoPauseDelay)
-		d.Set("collation", props.Collation)
-		d.Set("elastic_pool_id", props.ElasticPoolID)
-		if props.LicenseType != "" {
-			d.Set("license_type", props.LicenseType)
-		} else {
-			// value not returned, try to set from existing state/config
-			d.Set("license_type", d.Get("license_type").(string))
-		}
-		if props.MaxSizeBytes != nil {
-			d.Set("max_size_gb", int32((*props.MaxSizeBytes)/int64(1073741824)))
-		}
-		d.Set("min_capacity", props.MinCapacity)
-		d.Set("read_replica_count", props.HighAvailabilityReplicaCount)
-		if props.ReadScale == sql.DatabaseReadScaleEnabled {
-			d.Set("read_scale", true)
-		} else if props.ReadScale == sql.DatabaseReadScaleDisabled {
-			d.Set("read_scale", false)
-		}
-		if props.CurrentServiceObjectiveName != nil {
-			skuName = *props.CurrentServiceObjectiveName
-		}
-		d.Set("sku_name", skuName)
-		d.Set("storage_account_type", string(props.CurrentBackupStorageRedundancy))
-		d.Set("zone_redundant", props.ZoneRedundant)
-		if props.IsLedgerOn != nil {
-			ledgerEnabled = *props.IsLedgerOn
+
+	if model := resp.Model; model != nil {
+		if model.Name != nil {
+			d.Set("name", id.DatabaseName)
 		}
```

Comment:
`name` comes from the ID so isn't reliant on the field being returned from the API

```suggestion
	d.Set("name", id.DatabaseName)
	if model := resp.Model; model != nil {
```

Reply:
Fixed.